source/ingress.go

```diff
@@ -129,6 +129,10 @@ func (sc *ingressSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, e
 
 		ingEndpoints := endpointsFromIngress(ing, sc.ignoreHostnameAnnotation, sc.ignoreIngressTLSSpec, sc.ignoreIngressRulesSpec)
 
+		if sc.fqdnTemplate != nil && endpoint.EndpointsHaveEmptyTargets(ingEndpoints) {
```

Comment:
Another comment maybe:
```suggestion
                // Template configurations are a fallback to determine endpoint targets 
		if sc.fqdnTemplate != nil && endpoint.EndpointsHaveEmptyTargets(ingEndpoints) {
```